vms/platformvm/state/diff.go

```diff
@@ -194,6 +198,70 @@ func (d *diff) DeleteExpiry(entry ExpiryEntry) {
 	d.expiryDiff.DeleteExpiry(entry)
 }
 
+func (d *diff) GetActiveSubnetOnlyValidatorsIterator() (iterator.Iterator[SubnetOnlyValidator], error) {
+	parentState, ok := d.stateVersions.GetState(d.parentID)
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", ErrMissingParentState, d.parentID)
+	}
+
+	parentIterator, err := parentState.GetActiveSubnetOnlyValidatorsIterator()
+	if err != nil {
+		return nil, err
+	}
+
+	return d.sovDiff.getActiveSubnetOnlyValidatorsIterator(parentIterator), nil
+}
+
+func (d *diff) NumActiveSubnetOnlyValidators() int {
+	return d.parentActiveSOVs + d.sovDiff.numAddedActive
+}
+
+func (d *diff) WeightOfSubnetOnlyValidators(subnetID ids.ID) (uint64, error) {
+	if weight, modified := d.sovDiff.modifiedTotalWeight[subnetID]; modified {
+		return weight, nil
+	}
+
+	parentState, ok := d.stateVersions.GetState(d.parentID)
```

Comment:
can you explain the high level idea of doing this backwards traversal in search of the subnet, both here and in the next function? 

We basically don't have a full snapshot of the validators in each state? Why is that?

Reply:
Performing a full copy of the validator states for each diff would be pretty expensive. The number of changes per block is typically small, but the number total entries being tracked can be quite large.